spec/datadog/core/telemetry/logging_spec.rb

```diff
@@ -82,6 +82,126 @@ def log!(_event)
         end
       end
     end
+    context 'with an Erro error carrying telemetry message' do
+      subject(:report) do
+        raise Errno::ENOENT, 'Dynamic runtime message'
+      rescue SystemCallError => error
+        # This is normally done by native extensions, when raising Errno errors
+        error.instance_variable_set(:@telemetry_message, 'Safe message')
+
+        component.report(error, level: :error, description: description)
+      end
+
+      let(:description) { nil }
+
+      it 'includes the telemetry message but not the dynamic exception message' do
+        expect(component).to receive(:log!).with(instance_of(Datadog::Core::Telemetry::Event::Log)) do |event|
+          expect(event.payload).to include(
+            logs: [{message: 'Errno::ENOENT: (Safe message)', level: 'ERROR', count: 1,
+                    stack_trace: a_string_including('REDACTED')}]
+          )
+          expect(event.payload[:logs].first[:message]).not_to include('Dynamic runtime message')
+        end
+
+        report
+      end
+
+      context 'with description' do
+        let(:description) { 'Operation failed' }
+
+        it 'includes both description and telemetry message' do
+          expect(component).to receive(:log!).with(instance_of(Datadog::Core::Telemetry::Event::Log)) do |event|
+            expect(event.payload).to include(
+              logs: [{message: 'Errno::ENOENT: Operation failed (Safe message)', level: 'ERROR', count: 1,
+                      stack_trace: a_string_including('REDACTED')}]
+            )
+          end
+
+          report
+        end
+      end
+    end
+    context 'with NativeError' do
+      before do
+        skip unless defined?(Datadog::Core::Native::Error)
+      end
+      it 'includes the telemetry-safe message in telemetry' do
+        expect(component).to receive(:log!).with(instance_of(Datadog::Core::Telemetry::Event::Log)) do |event|
+          expect(event.payload).to include(
+            logs: [{message: 'Datadog::Core::Native::RuntimeError: (This is a safe profiler error)', level: 'ERROR', count: 1,
+                    stack_trace: a_string_including('REDACTED')}]
+          )
+        end
+        begin
+          raise Datadog::Core::Native::RuntimeError.new('This is a safe profiler error', 'This is a safe profiler error')
+        rescue => e
+          component.report(e, level: :error)
+        end
+      end
+      context 'with description' do
+        it 'includes both description and telemetry message' do
+          expect(component).to receive(:log!).with(instance_of(Datadog::Core::Telemetry::Event::Log)) do |event|
+            expect(event.payload).to include(
+              logs: [{message: 'Datadog::Core::Native::RuntimeError: Profiler failed to start (Failed to initialize native extension)', level: 'ERROR', count: 1,
+                      stack_trace: a_string_including('REDACTED')}]
+            )
+          end
```

Comment:
Should we assert on the `description`? (Similar to above?)

Reply:
I've changed it to assert that the dynamic exception message is not present. I'm assuming that's what you meant, since the description is already being matched in `message: 'Datadog::Core::Native::RuntimeError: Profiler failed to start (Failed to initialize native extension)'` (the description is `Profiler failed to start`).